src/models/index.js

```diff
@@ -18,11 +18,10 @@ Model.prototype.pull = function (req, callback) {
 }
 
 function createKey (req) {
-  let key = (req.params.layer)
-    ? req.url.split('/')[1] + '_' + req.params.layer
-    : req.url.split('/')[1]
+  let key = req.url.split('/')[1]
   if (req.params.host) key = [key, req.params.host].join('::')
   if (req.params.id) key = [key, req.params.id].join('::')
+  if (req.params.layer) key = [key, '_', req.params.layer].join('::')
```

Comment:
You don't need the `_` here as we are already using `::` as a delimiter.

`[key, req.params.layer].join('::')` alone will give you a key like `<provider>::<host>::<id>::<layer>`.